src/common_types.hpp

```diff
@@ -53,6 +53,18 @@ DEFINE_LEXICOGRAPHIC_ORDERING(cell_member_type,(a.gid,a.index),(b.gid,b.index))
 
 using time_type = float;
 
+// Enumeration used to indentify the cell type/kind, used by the model to
+// group equal kinds in the same cell group.
+
+enum cell_kind {
+    multicompartment,
+    poisson,
+    from_file,
```

Comment:
I was thinking about a neuron injecting spikes from file. This would allow deterministic testing. It is a usable cell type on the same level as the Poisson spike, I should be more clear in the enum naming? 

From slack: sam [4:35 PM] 
Just thinking about names. 'Multicompartment' is a bit of a misnomer, because we're really describing a cable-model neuron which is continuous in space; the compartmentalization is distinct from every other aspect of its description. 'Poisson' is also a bit of a misnomer, because as I understand it, these spike sources will produce a number of different sorts of random spike chains, only one of which is actually Poisson.

We have to use the term 'multicompartment' when describing NestMC because that is the term that describes this class of simulator. Internally though, it would be nice to be able to make a distinction between the continuous description and the discretization.

[4:38] 
I'd lean more towards names such as `point_neuron` (or perhaps `single_compartment`); `cable1d_neuron`; `stochastic_spike_source`; etc.

w.klijn [4:39 PM] 
Good call. It is taken from NEST but we can go our own 'logical' way in naming things.

[4:39] 
Ow: the from file neuron would replay spikes from a file on disk.

sam [4:39 PM] 
Then `spike_replay` :smile: